src/main/kotlin/no/nav/familie/tilbake/service/dokumentbestilling/vedtak/VedtaksbrevFritekstMapper.kt

```diff
@@ -0,0 +1,110 @@
+import java.util.UUID
+
+object VedtaksbrevFritekstMapper {
+
+    fun tilDomene(behandlingId: UUID, oppsummeringstekst: String?): Vedtaksbrevsoppsummering {
+        return Vedtaksbrevsoppsummering(behandlingId = behandlingId,
+                                        oppsummeringFritekst = oppsummeringstekst)
+    }
+
+    fun tilDomeneVedtaksbrevsperiode(behandlingId: UUID,
+                                     perioderMedFritekst: List<PeriodeMedTekstDto>): List<Vedtaksbrevsperiode> {
+        val vedtaksbrevsperioder = mutableListOf<Vedtaksbrevsperiode>()
+        perioderMedFritekst.forEach {
+            lagFaktaAvsnitt(behandlingId, it)?.let { faktaAvsnitt -> vedtaksbrevsperioder.add(faktaAvsnitt) }
+            lagForeldelseAvsnitt(behandlingId, it)?.let { foreldelseAvsnitt -> vedtaksbrevsperioder.add(foreldelseAvsnitt) }
+            lagVilkårsvurderingAvsnitt(behandlingId, it)?.let { vilkårAvsnitt -> vedtaksbrevsperioder.add(vilkårAvsnitt) }
+            lagSærligGrunnerAvsnitt(behandlingId,
+                                    it)?.let { særligGrunnerAvsnitt -> vedtaksbrevsperioder.add(særligGrunnerAvsnitt) }
+            lagSærligGrunnerAnnetAvsnitt(behandlingId, it)?.let { annetAvsnitt -> vedtaksbrevsperioder.add(annetAvsnitt) }
+        }
+        return vedtaksbrevsperioder
+    }
+
+    fun mapFritekstFraDb(fritekstPerioder: List<Vedtaksbrevsperiode>): List<PeriodeMedTekstDto> {
+
+        val perioderTilMap = HashMap<Periode, MutableMap<Friteksttype, String>>()
+
+        fritekstPerioder.forEach {
+            val avsnittTilTekst = perioderTilMap.getOrDefault(it.periode, mutableMapOf())
+            avsnittTilTekst[it.fritekststype] = it.fritekst
+            perioderTilMap[it.periode] = avsnittTilTekst
+        }
+
+        return perioderTilMap.entries.map { (periode, avsnittTilTekst) ->
+            PeriodeMedTekstDto(periode = PeriodeDto(periode),
+                               faktaAvsnitt = avsnittTilTekst[Friteksttype.FAKTA],
+                               foreldelseAvsnitt = avsnittTilTekst[Friteksttype.FORELDELSE],
+                               vilkårAvsnitt = avsnittTilTekst[Friteksttype.VILKÅR],
+                               særligeGrunnerAvsnitt = avsnittTilTekst[Friteksttype.SÆRLIGE_GRUNNER],
+                               særligeGrunnerAnnetAvsnitt = avsnittTilTekst[Friteksttype.SÆRLIGE_GRUNNER_ANNET])
+        }
+    }
+
+
+    private fun lagFaktaAvsnitt(behandlingId: UUID, faktaAvsnittMedPeriode: PeriodeMedTekstDto): Vedtaksbrevsperiode? {
+        if (faktaAvsnittMedPeriode.faktaAvsnitt != null) {
+            return Vedtaksbrevsperiode(behandlingId = behandlingId,
+                                       periode = Periode(faktaAvsnittMedPeriode.periode),
+                                       fritekst = faktaAvsnittMedPeriode.faktaAvsnitt,
+                                       fritekststype = Friteksttype.FAKTA
+            )
+        }
+        return null
+    }
```

Comment:
Bruk let på disse metodene også 
return faktaAvsnittMedPeriode.faktaAvsnitt?.let{ 
Vedtaksbrevsperiode(behandlingId = behandlingId,
                                       periode = Periode(faktaAvsnittMedPeriode.periode),
                                       fritekst = it,
                                       fritekststype = Friteksttype.FAKTA
}